airbyte-integrations/connectors/source-coda/source_coda/source.py

```diff
@@ -0,0 +1,273 @@
+        return "categories"
+
+
+
+
+class Pages(CodaStream):
+
+    primary_key = "PAGES"
+
+    def __init__(self, doc_id, **kwargs):
+        super().__init__(**kwargs)
+        self._doc_id = doc_id
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"docs/{self._doc_id}/pages"
+
+class Tables(CodaStream):
+
+    primary_key = "TABLES"
+
+    def __init__(self, doc_id, **kwargs):
+        super().__init__(**kwargs)
+        self._doc_id = doc_id
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"docs/{self._doc_id}/tables"
+
+
+
+class Formulas(CodaStream):
+
+    primary_key = "FORMULAS"
+
+    def __init__(self, doc_id, **kwargs):
+        super().__init__(**kwargs)
+        self._doc_id = doc_id
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"docs/{self._doc_id}/formulas"
+
+class Controls(CodaStream):
+
+    primary_key = "CONTROLS"
+
+    def __init__(self, doc_id, **kwargs):
+        super().__init__(**kwargs)
+        self._doc_id = doc_id
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"docs/{self._doc_id}/controls"
+
+
+
+
+
+
+
+# Basic incremental stream
+class IncrementalCodaStream(CodaStream, ABC):
+    """
+    TODO fill in details of this class to implement functionality related to incremental syncs for your connector.
+         if you do not need to implement incremental sync for any streams, remove this class.
+    """
+
+    # TODO: Fill in to checkpoint stream reads after N records. This prevents re-reading of data if the stream fails for any reason.
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> str:
+        """
+        TODO
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        return {}
```

Comment:
haven't yet implemented it, is it required for the creation of a data source. It could probably better if it's done in another pr, incremental feature addition ;)

Reply:
Remove it if isn't used.